iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/optional.inl

```diff
@@ -288,7 +276,7 @@ template <typename T>
 template <typename... Targs>
 inline void optional<T>::construct_value(Targs&&... args) noexcept
 {
-    new (static_cast<T*>(static_cast<void*>(&m_data))) T(std::forward<Targs>(args)...);
+    new (&m_data) T(std::forward<Targs>(args)...);
```

Comment:
Why was the cast removed? Because of an Axivion warning? I'm not sure if passing an unaligned pointer to operator new is working on all platforms, see [here](https://stackoverflow.com/a/11782277/18276160).